### ISSUE TITLE
Circular event firing when removing feature on editable:drawing:cancel

### ISSUE THREAD

**Issue by Sequoia** [Leaflet/Leaflet.Editable#76]
I have various buttons that start the creation of different marker types:
```js
function pointButtonClickHandler(e){
  const marker = map.editTools.startMarker(null, {
    icon : getIconFor(e.target.value); //different icons for different buttons
  });
  marker.type = e.target.value; // We'll need this type for data extraction later
}
```

If you click a button, it starts edit of that type, & you can click on the map to place the marker. If you click one button, then you click *another* button, calling `startMarker` eventually calls `cancelDrawing` which fires `editable:drawing:cancel`:
```
onCancelDrawing (Leaflet.Editable.js:531)
cancelDrawing (Leaflet.Editable.js:550)
unregisterForDrawing (Leaflet.Editable.js:128)
registerForDrawing (Leaflet.Editable.js:110)
startDrawing (Leaflet.Editable.js:540)
startMarker (Leaflet.Editable.js:163)
pointButtonClickHandler (mapEvents.js:154)
```

## The Initial Problem

While attempting to pick a new point type while another is waiting to be placed cancels drawing of the "floating" marker, **it does not remove it from the `featuresLayer`**. That means the point (that was never actually placed) is just dropped wherever you moused-out of the map.

### Solution:

When `editable:drawing:cancel` is fired, remove the marker from the `featuresLayer`:
```js
map.on('editable:drawing:cancel', function(e){
  console.log(`removing ${e.layer._leaflet_id}`);
  map.featuresLayer.removeLayer(e.layer);
});
```

Result:
```
removing 37
removing 37
leaflet-src.js:3623 Uncaught TypeError: Failed to execute 'removeChild' on 'Node': parameter 1 is not of type 'Node'.
```

## Secondary Problem

Because `enableDrawing` [attaches `disable` to the `remove` event](https://github.com/Leaflet/Leaflet.Editable/blob/leaflet0.7/src/Leaflet.Editable.js#L488) and [`disable` calls `cancelDrawing`](https://github.com/Leaflet/Leaflet.Editable/blob/leaflet0.7/src/Leaflet.Editable.js#L498), **removing the layer from `featuresLayer` on `editable:drawing:cancel` causes  `editable:drawing:cancel` to be fired again**.

## Workaround
Do your own check of the layer state before attempting to remove it:
```
map.on('editable:drawing:cancel', function(e){
  if(e.layer.editEnabled()){
    console.log(`removing ${e.layer._leaflet_id}`);
    map.featuresLayer.removeLayer(e.layer);
  }
});
```

*(there are other workarounds but this is the best I found)*.

## Request

There should be some simple way to remove a layer from the `featuresLayer` when editing is cancelled without triggering a second `editable:drawing:cancel` event. If this plugin [automatically adds stuff to the `featuresLayer`](https://github.com/Leaflet/Leaflet.Editable/blob/leaflet0.7/src/Leaflet.Editable.js#L136-L138), it should automatically remove stuff on cancel, or provide some clean way of doing so.

Thanks!

**Comment by Sequoia** [comment]
Oddly, this does *not* work:
```js
marker.once('editable:drawing:cancel', function(e){
    map.featuresLayer.removeLayer(e.layer);
});
```

it fires **twice** not once, and the issue persists.

**Comment by yohanboniface** [comment]
Which version are you using? This should be fixed in the 1.0.0-rc1 release AFAIK.

**Comment by Sequoia** [comment]
`0.7`. I guess it's fixed then!
